src/main/java/org/hisp/dhis/BaseDhis2.java

```diff
@@ -366,10 +362,36 @@ protected URI getObjectQuery(URIBuilder uriBuilder, Query query) {
   protected <T> T getDataValueSetResponse(
       URIBuilder uriBuilder, DataValueSetQuery query, Class<T> type) {
     URI url = getDataValueSetQuery(uriBuilder, query);
+    return getObjectFromUrl(url, type);
+  }
 
+  /**
+   * Retrieves a Validation object using HTTP GET.
+   *
+   * @param uriBuilder the URI builder.
+   * @param query the {@link DataSetValidationQuery} filters to apply.
+   * @param type the class type of the object.
+   * @param <T> type.
+   * @return the object.
+   */
+  protected <T> T getDataSetValidationResponse(
+      URIBuilder uriBuilder, DataSetValidationQuery query, Class<T> type) {
+    URI url = getDataSetValidationQuery(uriBuilder, query);
     return getObjectFromUrl(url, type);
```

Comment:
Just wondering, why do we need the generic type `T` here? It seems we only call this method with `Validation.class`. Perhaps just use `Validation` as type?